### PR TITLE
Remove verbose test coverage stats when tests fail

### PR DESCRIPTION
**Pull request by seanpreston**
### Description of Changes

The `--cov` pytest options cause a long code coverage report to be output every time a test fails, leaving whoever is attempting to read the test failures an unnecessarily long way to scroll to see the failure itself. An example can be found is [this gist](https://gist.github.com/seanpreston/fb3c040ab9cce2a2caece934849a5e99). With these options turned off the output is much easier to read:

```
root@78398bdf88be:/fides# pytest tests/lib/test_oauth_util.py 
===================================================== test session starts =====================================================
platform linux -- Python 3.10.7, pytest-7.1.2, pluggy-1.0.0 -- /usr/local/bin/python
cachedir: .pytest_cache
rootdir: /fides, configfile: pyproject.toml
plugins: requests-mock-1.10.0, celery-0.0.0, anyio-3.6.2, asyncio-0.19.0, env-0.6.2, Faker-14.1.0, cov-4.0.0
asyncio: mode=auto
collected 11 items                                                                                                            

tests/lib/test_oauth_util.py::test_jwe_create_and_extract PASSED                                                        [  9%]
tests/lib/test_oauth_util.py::test_is_token_expired[issued_at0-1-True] PASSED                                           [ 18%]
tests/lib/test_oauth_util.py::test_is_token_expired[issued_at1-60-False] PASSED                                         [ 27%]
tests/lib/test_oauth_util.py::test_is_token_expired[None-60-True] PASSED                                                [ 36%]
tests/lib/test_oauth_util.py::test_verify_oauth_malformed_oauth_client FAILED                                           [ 45%]
tests/lib/test_oauth_util.py::test_verify_oauth_client_no_issued_at PASSED                                              [ 54%]
tests/lib/test_oauth_util.py::test_verify_oauth_client_expired PASSED                                                   [ 63%]
tests/lib/test_oauth_util.py::test_verify_oauth_client_no_client_id PASSED                                              [ 72%]
tests/lib/test_oauth_util.py::test_verify_oauth_client_no_client PASSED                                                 [ 81%]
tests/lib/test_oauth_util.py::test_verify_oauth_client_wrong_security_scope PASSED                                      [ 90%]
tests/lib/test_oauth_util.py::test_verify_oauth_client_wrong_client_scope PASSED                                        [100%]

========================================================== FAILURES ===========================================================
__________________________________________ test_verify_oauth_malformed_oauth_client ___________________________________________

db = <sqlalchemy.orm.session.ExtendedSession object at 0x7f0d1161cbb0>
config = FidesConfig(test_mode=True, is_test_mode=True, hot_reloading=False, dev_mode=True, oauth_instance=None, admin_ui=Admin...et_length_bytes=16), user=UserSettings(auth_header=None, analytics_opt_out=True, encryption_key='test_encryption_key'))

    async def test_verify_oauth_malformed_oauth_client(db, config):
>       assert False
E       assert False

tests/lib/test_oauth_util.py:49: AssertionError
=================================================== short test summary info ===================================================
FAILED tests/lib/test_oauth_util.py::test_verify_oauth_malformed_oauth_client - assert False
================================================ 1 failed, 10 passed in 2.06s ================================================
```

### Code Changes

* [ ] remove `--cov` options from `pytest` defaults

### Questions

* [ ] Does anyone (or anything) use these coverage outputs? My expectation is that the answer is no, since they only show when tests fail.

### CODE REVIEW

**Comment by sanders41** [comment]
With this removed the coverage report doesn’t get generated and therefore isn’t available to upload to codecov. I suppose we could leave it out of the `pyproject.toml` and add it as a flag to each CI pytest job.

**Comment by sanders41** [comment]
@seanpreston I added the flags. Assuming coverage does show up after this we should be good to go.

**Comment by sanders41** [comment]
Looks like it ran coverage, but didn't generate the file 😕 . I'll look into why.

**Comment by sanders41** [comment]
@seanpreston I got things working with coverage. I tried running unsafe tests just to make sure things still run there. Several of them failed, but that happens more often than they pass so I don't think it is related to the PR.

Since I ended up making several updates I'm not sure that I should be the one to approve.

**Comment by ThomasLaPiana** [comment]
@sanders41 any reason not to use the `--no-cov-on-fail` flag? https://pytest-cov.readthedocs.io/en/latest/config.html#reference

I've definitely used that in PRs before when testing locally....

**Comment by ThomasLaPiana** [comment]
> I don't have a preference on way or the other so I went ahead and made the change to `--no-cov-on-fail`

Seems cleaner/easier in this case, thank you! 🙂

File: noxfiles/run_infrastructure.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def _run_tests(
     _run_cmd_or_err(
         f'echo "running pytest for conditions: {pytest_path} with environment variables: {environment_variables}"'
     )
-    coverage_arg = "--cov-report=lcov:coverage/ops-integration.lcov"
+    coverage_arg = "--cov-report=xml"
```
Comment (ThomasLaPiana) [inline noxfiles/run_infrastructure.py:238]: good catch! I knew that one would get forgotten at some point lol

**Review by ThomasLaPiana — APPROVED** [review]

File: noxfiles/run_infrastructure.py (hunk 1 above)
Comment (sanders41) [reply noxfiles/run_infrastructure.py:238]: I would have sworn I did a global search for this initially and still somehow missed this one!

**Pull request merged**